zulip/integrations/bridge_with_irc/irc_mirror_backend.py

```diff
@@ -144,3 +144,6 @@ def on_dccchat(self, c: ServerConnection, e: Event) -> None:
             except ValueError:
                 return
             self.dcc_connect(address, port)
+
+    def on_error(self, c: ServerConnection, e: Event) -> None:
+        raise IRCError(e.target)
```

Comment:
Don’t steal someone else’s exception class like this. It’s intended to be raised by the library, so its constructor interface belongs to the library and could change at the library’s discretion. For example, its constructor could gain more required arguments, like the corresponding `irc.server.IRCError` [already has](https://github.com/jaraco/irc/blob/v20.3.0/irc/server.py#L71).